crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -763,38 +870,28 @@ impl<'a> IRGenerator<'a> {
         env: &mut Environment,
         access: HirMemberAccess,
     ) -> Result<Value, RuntimeError> {
-        match self.expression_to_object(env, &access.lhs)? {
-            Value::Single(_) => unreachable!(
-                "Runtime type error, expected struct but found a single value for {:?}",
-                access
-            ),
-            Value::Struct(fields) => {
-                let field = dbg!(fields)
-                    .into_iter()
-                    .find(|(field_name, _)| *field_name == access.rhs.0.contents);
-
-                Ok(field.unwrap().1)
-            }
-        }
+        let value = self.expression_to_object(env, &access.lhs)?;
+        Ok(value.get_field_member(&access.rhs.0.contents).clone())
     }
 
-    //TODO generate phi instructions
     pub fn expression_list_to_objects(
         &mut self,
         env: &mut Environment,
         exprs: &[ExprId],
     ) -> Vec<NodeId> {
-        exprs
-            .iter()
-            .map(|expr| {
-                match self.expression_to_object(env, expr) {
-                    Ok(Value::Single(id)) => id,
-                    // TODO: Can we have arrays of structs? How should we store each element if
-                    // structs don't exist in ssa?
-                    other => panic!("Unexpected {:?} while codegening ssa array elements", other),
+        let mut result = Vec::new();
+        for expr in exprs {
+            match self.expression_to_object(env, expr) {
+                Ok(Value::Single(id)) => result.push(id),
+                Ok(Value::Struct(fields)) => {
+                    for f in fields {
+                        result.extend(f.1.to_node_ids());
```

Comment:
Flattening the nested ids here could result in having multiple keys for a given field name, we should try to keep the same Value structure `Vec<Value>`. That way when a block is used as a return value we will correctly extract all the fields of the last value and only those fields

Reply:
The function is just used for arrays and call arguments which both use only NodeIds. There is no point to add a field name which will not be used.

Reply:
Done

Reply:
Looks good. In the future, we should prefer `append` over `extend` for one Vec to another since the former is more efficient and they're otherwise equal